README.adoc

```diff
@@ -552,6 +552,32 @@ validates_length_of :email, maximum: 100
 validates :email, presence: true, length: { maximum: 100 }
 ----
 
+=== Custom Validation Methods
+
+When naming custom validation methods, adhere to the simple rules:
+
+ - `validate :method_name` reads like a natural statement
+ - the method name explains what it checks
+ - the method is recognizable as a validation method by its name, not a predicate method
+
+[source,ruby]
+----
+# good
+validate :expiration_date_cannot_be_in_the_past
+validate :discount_cannot_be_greater_than_total_value
+validate :ensure_same_topic
+
+# also good - explicit prefix
+validate :validate_birthday_in_past
+validate :validate_sufficient_quantity
+validate :must_have_owner_with_no_other_items
+validate :must_have_shipping_units
+
+# bad
+validate :birthday_in_past
```

Comment:
Neither of these seem that bad... can we find some worse examples? 😁

Reply:
The worst custom validation method names I could find:
```ruby
validate :final_grader_ok?
validate :valid_role?
validate :positive_points_possible?
validate :published_article
validate :users_for_shared_access
validate :close_after_open
validate :orphaned_annotation
validate :not_from_fork
validate :one_issue_tracker
validate :first_default_scope
validate :hosts_format
validate :permissions
```

Some of them read very well in conjunction with `validate'. But method names themselves don't give a clue that it's a custom validation method.

Which of them do you dislike most of all?

Reply:
The issue that I was raising in #223  was not that the validation methods read poorly when being called as part of the `validate :xyz` line, but rather that reading through the class's code doesn't make it immediately obvious if the method is a validation method or a predicate method (or something else). That's why I personally have adopted the `validate :validate_xyz` pattern, even though admittedly it reads somewhat poorly.